### PR TITLE
Drag To Select Table

### PR DESCRIPTION
**Pull request by brendanlaschke**
POC of how drag to select on the tables could be implemented. This should not be merged as this is not quite production ready :) 

Some points to discuss:
* How to handle previously made selections
* Are rows additionally marked to the checkmark in the first column?
* How to handle if some selections where made previously -> drag over but don't release -> drag back - should they retain the selected state ?
* and maybe more ... 

@charlesBochet @FelixMalfait What are your thoughts on this?

### CODE REVIEW

**Comment by charlesBochet** [comment]
@brendanlaschke This looks promising. Could you add a video of the current behavior?

Open Questions:
- can we style the selection box?

Things to consider while building the feature:
- think about the dev experience: maybe we want to move this into a customHook that's easily applicable to table and pipelines. Maybe we want one for tables and one for pipelines and they are different
- be careful with performances here: while dragging the selection we want to avoid re-rendering (you can use React Component developer plugin to check that)

If we can get a good dev xp, customizble design and good performance, we are happy with it!

(I'm also surprised by the changes on the yarn.lock, something seems wrong!)

File: front/src/modules/ui/table/components/EntityTableDragSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      );
+    },
+    selectionProps: {
+      style: {
+        border: '1px solid #4C85D8',
+        background: 'rgba(155, 193, 239, 0.4)',
+        position: `absolute`,
+        zIndex: 99,
+      },
+    },
```
Comment (brendanlaschke) [inline front/src/modules/ui/table/components/EntityTableDragSelect.tsx:61]: Styling would be possible here

**Comment by brendanlaschke** [comment]
https://github.com/twentyhq/twenty/assets/48770548/53c71c66-4912-4ffb-b5cc-786b52e6dc76

File: front/src/modules/ui/table/components/EntityTableDragSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  boxesIntersect,
+  useSelectionContainer,
+} from '@air/react-drag-to-select';
+import { useRecoilCallback, useRecoilValue } from 'recoil';
+
+import { isRowSelectedFamilyState } from '../states/isRowSelectedFamilyState';
+import { tableRowIdsState } from '../states/tableRowIdsState';
+
+type OwnProps = {
+  tableBodyRef: RefObject<HTMLDivElement>;
```
Comment (charlesBochet) [inline front/src/modules/ui/table/components/EntityTableDragSelect.tsx:12]: can we abstract this a little bit more to uncouple from table:
- dragSelectable: ref
- onDragSelect: (id: string) => void

selectable would be used to:
- set eventsElement to be dragSelectable itself
- compute the list of selectable items: all selectable children that have a 'selectable-id' attribute

onDragSelect would be called with the selectable-id as attribute.

We can also find a more generic name for EntityTableDragSelect. Maybe just DragSelect and move this into @/ui/utils/drag-select

Then we can use it from EntityTable

**Review by charlesBochet — COMMENTED** [review]
Very excited about it. 
Looks good for styling, I've made a proposal to give a more abstract UI in comment.
We then only need to double check the performances are we are good to go!

**Comment by brendanlaschke** [comment]
I made some changes - hope as you pictured them above :) 

Before merge the yarn.lock file should be checked if it is correct (and maybe generated by one of you) - it seems like my setup generates weird yarn.lock files ??

**Comment by charlesBochet** [comment]
This is super weird! What version of node are you using? 
I've merged: https://github.com/twentyhq/twenty/pull/1083, so you can rebase on twenty/main to get the dependency

**Comment by charlesBochet** [comment]
I have merged main into your branch here actually :)

**Review by charlesBochet — APPROVED** [review]
Bravo!

**Comment by brendanlaschke** [comment]
> This is super weird! What version of node are you using?
> I've merged: #1083, so you can rebase on twenty/main to get the dependency

Im on an M1 Mac with NodeJs v18.6 and yarn 3.6 ...

**Pull request merged**